uniond/app/upgrades/v1_2_0/upgrade.go

```diff
@@ -0,0 +1,240 @@
+package v1_2_0
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+
+	"cosmossdk.io/math"
+
+	upgradetypes "cosmossdk.io/x/upgrade/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	crisistypes "github.com/cosmos/cosmos-sdk/x/crisis/types"
+	distributiontypes "github.com/cosmos/cosmos-sdk/x/distribution/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	minttypes "github.com/cosmos/cosmos-sdk/x/mint/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+
+	"github.com/unionlabs/union/uniond/app/upgrades"
+)
+
+const U_BASE_DENOM = "au"
+
+// One U in `au`
+const ONE_U = 1_000_000_000_000_000_000
+
+// Total supply of U (note, not in `au`)
+const U_TOTAL_SUPPLY = 10_000_000_000
+
+// Union foundation multisig address
+const UNION_FOUNDATION_MULTI_SIG = "union1cpz5fhesgjcv2q0640uxtyur5ju65av6r8fem0" // NOTE: Must add branch logic for mainnet before running migration
```

Comment:
on second thought, maybe it'd be best to keep these in a map indexed by chain id?